src/states/bootstrapping_peer.rs

```diff
@@ -123,24 +123,31 @@ impl BootstrappingPeer {
         let token = self.timer.schedule(BOOTSTRAP_TIMEOUT);
         let _ = self.timeout_tokens.insert(token, dst.peer_addr);
 
-        // If we are relocating, request bootstrap to the section matching the name given to us
-        // by our section. Otherwise request bootstrap to the section matching our current name.
-        let destination = if let Some(details) = self.relocate_details.as_ref() {
+        let destination = self.get_destination();
+        self.send_direct_message(&dst, DirectMessage::BootstrapRequest(destination));
+        self.peer_map_mut().connect(dst);
+    }
+
+    // If we are relocating, request bootstrap to the section matching the name given to us
+    // by our section. Otherwise request bootstrap to the section matching our current name.
+    fn get_destination(&self) -> XorName {
+        if let Some(details) = self.relocate_details.as_ref() {
             details.content().destination
         } else {
             *self.name()
-        };
-
-        self.send_direct_message(&dst, DirectMessage::BootstrapRequest(destination));
-        self.peer_map_mut().connect(dst);
+        }
     }
 
     fn join_section(&mut self, info: EldersInfo) -> Result<Transition, RoutingError> {
         let old_full_id = self.full_id.clone();
-        let prefix = info.prefix();
+        let destination = self.get_destination();
+
+        // Use a name that will match the destination even after multiple splits
+        let extra_split_count = 3;
```

Comment:
Nice, this should fix: https://github.com/maidsafe/routing/issues/1861.
However, it might be worth using non-constant number of bits - the longer the target prefix is, the fewer bits to use. This is because for long prefixes, generating the matching name might become too expensive.Also long prefixes are less likely to split.

Reply:
I'm not sure that long prefix are less likely to be split, and section for balancing need to be able to specify narrower interval in the future. I would probably not complicate that until we decide what to do?

Reply:
> I'm not sure that long prefix are less likely to be split,

The longer the prefix, the lower the probability that a random name ends up being matched by it and so lower probability that any node ends up joining it. 

But this can be addressed later. It's not important for this PR.